src/github/branch.ts

```diff
@@ -3,38 +3,102 @@ import issueKeyParser from "jira-issue-key-parser";
 import { emitWebhookProcessedMetrics } from "../util/webhooks";
 import { CustomContext } from "./middleware";
 import _ from "lodash";
-import {WebhookPayloadCreate, WebhookPayloadDelete} from "@octokit/webhooks";
+import { WebhookPayloadCreate, WebhookPayloadDelete } from "@octokit/webhooks";
+import { booleanFlag, BooleanFlags } from "../config/feature-flags";
+import sqsQueues from "../sqs/queues";
+import { GitHubAPI } from "probot";
+import { LoggerWithTarget } from "probot/lib/wrap-logger";
+import getJiraClient from "../jira/client";
 
 export const createBranch = async (
 	context: CustomContext,
-	jiraClient
+	jiraClient,
+	_util,
+	githubInstallationId: number
 ): Promise<void> => {
 
 	const webhookPayload: WebhookPayloadCreate = context.payload;
-	const jiraPayload = await transformBranch(context.github, webhookPayload);
 
-	if (!jiraPayload) {
+	if (await booleanFlag(BooleanFlags.USE_SQS_FOR_BRANCH, false, jiraClient.baseURL)) {
+		await sqsQueues.branch.sendMessage({
+			jiraHost: jiraClient.baseURL,
+			installationId: githubInstallationId,
+			webhookReceived: new Date().getTime(),
+			webhookId: context.id,
+			webhookPayload
+		})
+	} else {
+
+		const jiraPayload = await transformBranch(context.github, webhookPayload);
+
+		if (!jiraPayload) {
+			context.log(
+				{ noop: "no_jira_payload_create_branch" },
+				"Halting further execution for createBranch since jiraPayload is empty"
+			);
+			return;
+		}
+
 		context.log(
+			`Sending jira update for create branch event for hostname: ${jiraClient.baseURL}`
+		);
+
+		const jiraResponse = await jiraClient.devinfo.repository.update(jiraPayload);
+		const { webhookReceived, name, log } = context;
+
+		webhookReceived && emitWebhookProcessedMetrics(
+			webhookReceived,
+			name,
+			log,
+			jiraResponse?.status
+		);
+	}
+};
+
+export const processBranch = async (
```

Comment:
This function is a copy of the old processing logic (in the `else` block above), polished and modified to abstract away from Probot.